pkgs/by-name/cm/cmake/setup-hook.sh

```diff
@@ -45,21 +56,21 @@ cmakeConfigurePhase() {
     # Unfortunately cmake seems to expect absolute paths for ar, ranlib, and
     # strip. Otherwise they are taken to be relative to the source root of the
     # package being built.
-    cmakeFlags="-DCMAKE_CXX_COMPILER=$CXX $cmakeFlags"
-    cmakeFlags="-DCMAKE_C_COMPILER=$CC $cmakeFlags"
-    cmakeFlags="-DCMAKE_AR=$(command -v $AR) $cmakeFlags"
-    cmakeFlags="-DCMAKE_RANLIB=$(command -v $RANLIB) $cmakeFlags"
-    cmakeFlags="-DCMAKE_STRIP=$(command -v $STRIP) $cmakeFlags"
+    cmakeFlagsArray=("-DCMAKE_CXX_COMPILER=$CXX" "${cmakeFlagsArray[@]}")
```

Comment:
Shouldn't this be 
```suggestion
    cmakeFlagsArray+=("-DCMAKE_CXX_COMPILER=$CXX" "${cmakeFlagsArray[@]}")
```
?

Reply:
This is meant to prepend the flag to `cmakeFlagsArray` 

The prepended flag have lower precedence, allowing users to override the default behaviors with `cmakeFlags` and `cmakeFlagsArray`.

Reply:
Nevertheless, it's surely awkward to have to repeat `'${cmakeFlagsArray[@]}"` so many times. I did so to keep flag order the same as current implementation (where `$cmakeFlags` are prepended in a similar manner). Is it okay to define something like `__cmakeFlagsArrayDefault`, append to it, and then prepend it to `cmakeFlagsArray`? This will only affects the relative order of the flags provided by this setup hook, and their precedence will still be lower than the custom `cmakeFlags` and `cmakeFlagsArray` specification.

Reply:
> This is meant to prepend the flag to `cmakeFlagsArray`

Hum, understood.

> Nevertheless, it's surely awkward to have to repeat `'${cmakeFlagsArray[@]}"` so many times.

I have not linted my own code :(
Anyway my question was answered. TKS!

Reply:
@ShamrockLee for what it's worth, I'm very much in favor of something like `__cmakeFlagsArrayDefault`! Especially if it would allow us to group all these together, ahead of time.